tools/tests/py2fgen/conftest.py

```diff
@@ -9,8 +9,67 @@
 from pathlib import Path
 
 import pytest
+from icon4py.model.common.settings import xp
 
 
 @pytest.fixture
 def samples_path():
     return Path(__file__).parent / "fortran_samples"
+
+
+def compare_values_shallow(value1, value2, obj_name="value"):
+    # Compare if both are objects with attributes (__dict__)
+    if hasattr(value1, "__dict__") and hasattr(value2, "__dict__"):
+        if value1.__class__ != value2.__class__:
+            return False, f"Class mismatch for {obj_name}: {value1.__class__} != {value2.__class__}"
```

Comment:
This does not test whether the actual values of class members are the same or not. Do you think it is necessary to test the values? Or maybe I am thinking too much.

Reply:
I had some issues at first doing this recursively, so I went with a simpler implementation here. However an expanded check definitely makes this test much more useful so I implemented this in the next PR to merge after this one (dycore granule PR) https://github.com/C2SM/icon4py/pull/508

See this file: https://github.com/C2SM/icon4py/blob/2bd09991871e55739f811d95e691ff5972b27a6d/tools/tests/py2fgen/conftest.py

Reply:
You are right, I've extended the checks now.